File: atcoder/abc393a_poisonous_oyster.hs
-- vicfred
-- https://atcoder.jp/contests/abc393/tasks/abc393_a
-- implementation

import System.IO
import Data.List (words)

main :: IO ()
main = do
    -- Read a single line with two words
    input <- getLine
    
    -- Split input into two words
    let [s1, s2] = words input
    
    -- Check conditions and print corresponding values
    print $ if s1 == "fine" && s2 == "fine" then 4
            else if s1 == "fine" && s2 == "sick" then 3
            else if s1 == "sick" && s2 == "fine" then 2
            else 1

File: atcoder/abc393a_poisonous_oyster.rs
// vicfred
// https://atcoder.jp/contests/abc393/tasks/abc393_a
// implementation
use std::io;

fn main() {
    let mut input = String::new();
    
    // Read a single line with two words
    io::stdin().read_line(&mut input).unwrap();
    
    // Split input into two words and trim whitespace
    let words: Vec<&str> = input.trim().split_whitespace().collect();
    let s1 = words[0];
    let s2 = words[1];
    
    // Check conditions and print corresponding values
    if s1 == "fine" && s2 == "fine" {
        println!("4");
    } else if s1 == "fine" && s2 == "sick" {
        println!("3");
    } else if s1 == "sick" && s2 == "fine" {
        println!("2");
    } else if s1 == "sick" && s2 == "sick" {
        println!("1");
    }
}
